packages/amplify-cli/src/extensions/amplify-helpers/trigger-flow.ts

```diff
@@ -45,6 +45,50 @@ export const addTrigger = async triggerOptions => {
     triggerEventPath,
     skipEdit,
   } = triggerOptions;
+  /**
+   * The function template dir for this trigger, i.e.
+   * PostConfirmation/function-template-dir.
+   */
+  const sourceRoot = path.join(triggerDir, 'function-template-dir');
+  /**
+   * The function template dir for the default trigger template, i.e.
+   * _default/function-template-dir.
+   */
+  const defaultRoot = path.resolve(triggerDir, '..', '_default', 'function-template-dir');
+  /**
+   * A key-value map of input filenames and output filepaths.
+   */
+  const templateMap = {
+    'trigger-index.js': path.join('src', 'index.js'),
+    'package.json.ejs': path.join('src', 'package.json'),
+    'event.json': path.join('src', 'event.json'),
+  };
+  /**
+   * Hold a key-value map of input filepaths and output filenames.
+   */
+  const destMap = {};
+  /**
+   * For each templateFile (i.e. trigger-index.js), use
+   * {triggerDir}/../_default/{templateFile} if {triggerDir}/{templateFile}
+   * override does not exist.
+   */
+  const templateFiles = ['trigger-index.js', 'package.json.ejs', 'event.json'];
+  const sourceFiles = templateFiles.map(file => {
+    const defaultTemplate = path.resolve(defaultRoot, file);
+    const overrideTemplate = path.resolve(sourceRoot, file);
+    const templateToUse = fs.existsSync(overrideTemplate) ? overrideTemplate : defaultTemplate;
+    return path.relative(sourceRoot, templateToUse);
```

Comment:
It looks like `sourceFiles` here is a relative path, but in the previous version, it was just the file's basename. Is that intentional?

Reply:
Yes. The comments could be updated for clarity, but it resolves each file to the default template if an override isn't found, and then resolves it relative to `sourceRoot` for the following call:

https://github.com/aws-amplify/amplify-cli/blob/56c139d31ddaa45741191461699653f96ce353c4/packages/amplify-cli/src/extensions/amplify-helpers/trigger-flow.ts#L98-L103